packages/dd-trace/src/appsec/iast/analyzers/nosql-injection-mongodb-analyzer.js

```diff
@@ -119,29 +126,25 @@ class NosqlInjectionMongodbAnalyzer extends InjectionAnalyzer {
       const rangesByKey = {}
       const allRanges = []
 
-      iterateObjectStrings(value.filter, (val, nextLevelKeys) => {
+      iterateMongodbQueryStrings(value.filter, (val, nextLevelKeys) => {
         let ranges = getRanges(iastContext, val)
-        if (ranges?.length) {
-          const filteredRanges = []
-
+        if (ranges?.length === 1) {
           ranges = this._filterSecureRanges(ranges)
           if (!ranges.length) {
             this._incrementSuppressedMetric(iastContext)
+            return
           }
 
-          for (const range of ranges) {
-            if (this._isVulnerableRange(range)) {
-              isVulnerable = true
-              filteredRanges.push(range)
-            }
+          const range = ranges[0]
+          if (!this._isVulnerableRange(range, val)) {
+            return
           }
+          isVulnerable = true
 
-          if (filteredRanges.length > 0) {
-            rangesByKey[nextLevelKeys.join('.')] = filteredRanges
-            allRanges.push(...filteredRanges)
-          }
+          rangesByKey[nextLevelKeys.join('.')] = ranges
+          allRanges.push(range)
         }
-      }, [], 4)
```

Comment:
Why object depth analysis has changed from 4 to 10 (the default one on `iterateMongodbQueryStrings`)?